src/ert/shared/storage/key_utils.py

```diff
@@ -0,0 +1,199 @@
+from enum import Enum, auto
+from typing import List
+
+special_vars = [
+    "NAIMFRAC",
+    "NBAKFL",
+    "NBYTOT",
+    "NCPRLINS",
+    "NEWTFL",
+    "NEWTON",
+    "NLINEARP",
+    "NLINEARS",
+    "NLINSMAX",
+    "NLINSMIN",
+    "NLRESMAX",
+    "NLRESSUM",
+    "NMESSAGE",
+    "NNUMFL",
+    "NNUMST",
+    "NTS",
+    "NTSECL",
+    "NTSMCL",
+    "NTSPCL",
+    "ELAPSED",
+    "MAXDPR",
+    "MAXDSO",
+    "MAXDSG",
+    "MAXDSW",
+    "STEPTYPE",
+    "WNEWTON",
+]
+rate_vars = [
+    "OPR",
+    "OIR",
+    "OVPR",
+    "OVIR",
+    "OFR",
+    "OPP",
+    "OPI",
+    "OMR",
+    "GPR",
+    "GIR",
+    "GVPR",
+    "GVIR",
+    "GFR",
+    "GPP",
+    "GPI",
+    "GMR",
+    "WGPR",
+    "WGIR",
+    "WPR",
+    "WIR",
+    "WVPR",
+    "WVIR",
+    "WFR",
+    "WPP",
+    "WPI",
+    "WMR",
+    "LPR",
+    "LFR",
+    "VPR",
+    "VIR",
+    "VFR",
+    "GLIR",
+    "RGR",
+    "EGR",
+    "EXGR",
+    "SGR",
+    "GSR",
+    "FGR",
+    "GIMR",
+    "GCR",
+    "NPR",
+    "NIR",
+    "CPR",
+    "CIR",
+    "SIR",
+    "SPR",
+    "TIR",
+    "TPR",
+    "GOR",
+    "WCT",
+    "OGR",
+    "WGR",
+    "GLR",
+]
+
+seg_rate_vars = [
+    "OFR",
+    "GFR",
+    "WFR",
+    "CFR",
+    "SFR",
+    "TFR",
+    "CVPR",
+    "WCT",
+    "GOR",
+    "OGR",
+    "WGR",
+]
+
+
+class SummaryVarType(Enum):
+    RD_SMSPEC_INVALID_VAR = auto()
+    RD_SMSPEC_FIELD_VAR = auto()
+    RD_SMSPEC_REGION_VAR = auto()
+    RD_SMSPEC_GROUP_VAR = auto()
+    RD_SMSPEC_WELL_VAR = auto()
+    RD_SMSPEC_SEGMENT_VAR = auto()
+    RD_SMSPEC_BLOCK_VAR = auto()
+    RD_SMSPEC_AQUIFER_VAR = auto()
+    RD_SMSPEC_COMPLETION_VAR = auto()
+    RD_SMSPEC_NETWORK_VAR = auto()
+    RD_SMSPEC_REGION_2_REGION_VAR = auto()
+    RD_SMSPEC_LOCAL_BLOCK_VAR = auto()
+    RD_SMSPEC_LOCAL_COMPLETION_VAR = auto()
+    RD_SMSPEC_LOCAL_WELL_VAR = auto()
+    RD_SMSPEC_MISC_VAR = auto()
+
+    @staticmethod
+    def determine_var_type(var: str) -> "SummaryVarType":
+        if var in special_vars:
+            return SummaryVarType.RD_SMSPEC_MISC_VAR
+        default_case = {
```

Comment:
Inline with return statement if not used elsewhere? (or move down to return statement).